mne/io/curry/curry.py

```diff
@@ -0,0 +1,244 @@
+     and its CURRY version
+     """
+
+    # we don't use os.path.splitext to also handle extensions like .cdt.dpa
+    fname_base, ext = full_fname.split(".", maxsplit=1)
+
+    if 'cdt' in ext:
+        curry_vers = 8
+        for check_ext in [".cdt", ".cdt.dpa"]:
+            if not os.path.isfile(fname_base + check_ext):
+                raise FileNotFoundError("The following required file cannot be"
+                                        " found: %s. Please make sure it is "
+                                        "located in the same directory as %s."
+                                        % (fname_base + check_ext, full_fname))
+
+    else:
+        curry_vers = 7
+        for check_ext in [".dap", ".dat", ".rs3"]:
+            if not os.path.isfile(fname_base + check_ext):
+                raise FileNotFoundError("The following required file cannot be"
+                                        " found: %s. Please make sure it is "
+                                        "located in the same directory as %s."
+                                        % (fname_base + check_ext, full_fname))
+
+    return fname_base, curry_vers
+
+
+def _read_curry_info(fname_base, curry_vers):
+    #####################################
+    # read parameters from the param file
+
+    if curry_vers == 7:
+        file_extension = '.dap'
+    else:
+        file_extension = '.cdt.dpa'
+
+    var_names = ['NumSamples', 'NumChannels', 'NumTrials', 'SampleFreqHz',
+                 'TriggerOffsetUsec', 'DataFormat', 'SampleTimeUsec',
+                 'NUM_SAMPLES', 'NUM_CHANNELS', 'NUM_TRIALS', 'SAMPLE_FREQ_HZ',
+                 'TRIGGER_OFFSET_USEC', 'DATA_FORMAT', 'SAMPLE_TIME_USEC']
+
+    param_dict = dict()
+    with open(fname_base + file_extension) as fid:
+        for line in fid:
+            if any(var_name in line for var_name in var_names):
+                key, val = line.replace(" ", "").replace("\n", "").split("=")
+                param_dict[key] = val
+
+    if not len(param_dict) == 7:
+        raise KeyError("Some variables cannot be found in the parameter file.")
+
+    if "NumSamples" in param_dict:
+        n_samples = int(param_dict["NumSamples"])
+        # n_ch = int(param_dict["NumChannels"])
+        n_trials = int(param_dict["NumTrials"])
+        sfreq = float(param_dict["SampleFreqHz"])
+        offset = float(param_dict["TriggerOffsetUsec"]) / 1e6  # convert to s
+        time_step = float(param_dict["SampleTimeUsec"]) / 1e6
+        # data_format = param_dict["DataFormat"]
+
+    else:
+        n_samples = int(param_dict["NUM_SAMPLES"])
+        # n_ch = int(param_dict["NUM_CHANNELS"])
+        n_trials = int(param_dict["NUM_TRIALS"])
+        sfreq = float(param_dict["SAMPLE_FREQ_HZ"])
+        offset = float(param_dict["TRIGGER_OFFSET_USEC"]) / 1e6
+        time_step = float(param_dict["SAMPLE_TIME_USEC"]) / 1e6
+        # data_format = param_dict["DATA_FORMAT"]
```

Comment:
This looks a bit ugly. I'm not sure it's actually preferable, you could make it nicer, e.g.

```
n_samples = int(param_dict.get("NUM_SAMPLES", param_dict["NumSamples"]))
```

etc

Or you simply rewrite the dict ...

```
param_dict = {key.lower().replace("_", ""): value for key, value in param_dict.items()}
```